internal/common/cloudproviderapi/client.go

```diff
@@ -16,21 +18,31 @@ type Client struct {
 	client    *http.Client
 }
 
+const (
+	defaultRetries = 3
+)
+
 func NewClient(authToken string, rawAPIURL string, client *http.Client) (*Client, error) {
 	parsedAPIURL, err := url.Parse(rawAPIURL)
 	if err != nil {
 		return nil, fmt.Errorf("failed to parse Cloud Provider API url: %w", err)
 	}
 
+	if client == nil {
+		retryClient := retryablehttp.NewClient()
+		retryClient.RetryMax = defaultRetries
+		client = retryClient.StandardClient()
```

Comment:
should we add as well a request timeout by default?

Reply:
added